fix(jest): cannot set `preset` and `globals` config when using in a typescript project

riywo opened the pull request:
Some of `jestConfig` properties are not able to update through constructor
because of a bug by assigning hard coded default values.

This commit fixes the bug to assign hard coded values only if the same
properties are not set.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

mrgrain commented on src/typescript/typescript.ts:
```diff
@@ -429,12 +429,16 @@ export class TypeScriptProject extends NodeProject {
+    if (!jest.config.preset) {
+      jest.config.preset = "ts-jest";
+    }
+    if (!jest.config.globals?.["ts-jest"]?.tsconfig) {
+      jest.config.globals = {
+        "ts-jest": {
+          tsconfig: this.tsconfigDev.fileName,
+        },
+      };
```
The if statement and the assignment don't really match up. I reckon we should deep merge the object

riywo replied:
Ah, right. This breaks if there are some other options on `jest.config.globals`. Let me fix it.

Do we already have some deep merge libraries or functions in this project?

mrgrain replied:
There's a `deepMerge` in `util.ts`. I guess it works. 😅

mrgrain replied:
@riywo I'd love to see this fix merged. Is there anything else you need from me?

riywo replied:
Just done (including rebase). Check this change: https://github.com/projen/projen/compare/ff3a1fd619d1035cb8a170cb32e71903a7e5a554..8b6d45d1bd8a42b97f04ee42489356b7124c8a51

mrgrain reviewed (changes requested):
thanks for the fix. Your PR needs to include the license statement footer from the PR template (or see other PRs ).

riywo commented:
@mrgrain Ah, sorry! I created this PR by `gh pr create --fill` so that the PR template wasn't used. Updated the description by the footer.

mrgrain commented:
No worries! Really should have some automation for that. Although I'm surprised `gh pr` doesn't honor the template 🤔

riywo commented:
@mrgrain I addressed your comment. Check this diff as well https://github.com/projen/projen/compare/ff3a1fd619d1035cb8a170cb32e71903a7e5a554..8b6d45d1bd8a42b97f04ee42489356b7124c8a51

riywo commented:
Thank you!
